src/utils/AutoControlledComponent-slim.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable */
+import _ from 'lodash'
+import { Component } from 'react'
+
+const toDefault = (prop) => `default${prop[0].toUpperCase() + prop.slice(1)}`
+
+export default class AutoControlledComponent extends Component {
+  componentWillMount() {
+    if (super.componentWillMount) super.componentWillMount()
+    const { autoControlledProps } = this.constructor
+
+    this.state = _.transform(autoControlledProps, (res, prop) => {
```

Comment:
Wouldw be cool to get some comments around this. It took me a couple minutes to fully grok it especially due to the generically named to `toDefault`. Not at all a blocker, just a note.